src/widget.cpp

```diff
@@ -1643,6 +1643,23 @@ std::string widget::layout( const avatar &ava, unsigned int max_width, int label
                 // for debug keep track of each and width
                 debug_widths.append( string_format( "%s: %d,", wid.str(), cur_width ) );
 
+                const bool skip_pad_this = skip_pad || wid->has_flag( json_flag_W_NO_PADDING );
+                // Layout child in this column
+                const std::string txt = cur_child.layout( ava, skip_pad_this ? 0 : cur_width,
+                                        label_width, skip_pad_this );
+                // Store the resulting text for this column
+                std::vector<std::string> rows = string_split( txt, '\n' );
+
+                if( rows.size() > 1 && cur_child._style != "layout" ) {
+                    cur_width = 0;
+                    for( const std::string &row : rows ) {
+                        int w = utf8_width( row, true );
+                        if( w > cur_width ) {
+                            cur_width = w;
```

Comment:
@dseguin Consulting you again as the author of the relevant code...

The problem with all the widgets that take/took too much space was a trailing newline in the produced strings. I now relize that they have a purpose. If I am right, it is for allowing multiline widgets in column layouts to only take as much horizontal space as required.

For row widgets to the right of such a multiline widgets that "hang below" the widget's height, this approach causes mis-aligned indentation.

My current solution is to use the newline for calculations and printing, but remove it in the end. Further, I calculate the width required by multiline widgets explicitly during column layout. Seems to work so far when viewing in-game, but tests still fail. I even tried the first of the two failing test widgets, and it looks good in-game.

Another problem with my solution is that it would probably not shrink the widget's space if it is wrapped into another layout.

**Proposed solution**

So, I would propose to completely get rid of this special shrinkage handing for multiline widgets. One would then have to put the widget into wrapper layout with explicit width for the same effect, but that's just how is works for all other widgets.

Reply:
Moved the changes after implementing my proposal over to #59676, as I think we should handle and test this all in one go.